common/dynamicconfig/memory_client.go

```diff
@@ -71,11 +71,14 @@ func (d *MemoryClient) getValueLocked(key Key) []ConstrainedValue {
 
 func (d *MemoryClient) Subscribe(f ClientUpdateFunc) (cancel func()) {
 	d.lock.Lock()
-	defer d.lock.Unlock()
-
 	d.subscriptionIdx++
 	id := d.subscriptionIdx
 	d.subscriptions[id] = f
+	d.lock.Unlock()
```

Comment:
This change doesn't do anything? It's still unlocking at the same place. `f` is not called here